### PR TITLE
Add a basic Makefile

### PR DESCRIPTION
**Pull request by acelaya**
Part of https://github.com/hypothesis/wp-hypothesis/issues/40

Add a very basic `Makefile`, which follows the usual tasks we have in other projects.

The only exception are tasks around testing, as there are no tests yet. We can add tasks related with tests as part of adding the tests themselves.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	@echo "make format   Automatically format code"
+	@echo "make sure     Make sure that the formatter, linter, tests, etc all pass"
+
+.PHONY: dev
+dev: vendor
+	docker compose up
+
+.PHONY: lint
+lint: vendor
+	composer lint
```
Comment (robertknight) [inline Makefile:18]: Is there any kind of PHP version management needed when installing dependencies, or is it like npm in our JS projects where we can get by without requiring a specific Node/npm version?

File: README.md [hunk 2]
```diff
@@ -11,6 +11,13 @@ Stable versions are available on the [Hypothesis plugin page on WordPress.org](h
 5. Click Activate
 6. Visit your WordPress Settings > Hypothesis page to configure how it works on your site
 
+## Development
+
+1. Install `php` and the `dom` and `mbstring` extensions.
```
Comment (robertknight) [inline README.md:16]: Is there a reference page we could link to for how to install these extensions? Are they typically included by default in the main PHP releases?

File: README.md [hunk 3]
```diff
@@ -11,6 +11,13 @@ Stable versions are available on the [Hypothesis plugin page on WordPress.org](h
...
+2. [Download composer](https://getcomposer.org/download/), the PHP package manager.
```
Comment (robertknight) [inline README.md:17]:
```suggestion
2. [Download Composer](https://getcomposer.org/download/), the PHP package manager.
```

I'm following the capitalization that [Composer uses in their Getting Started guide](https://getcomposer.org/doc/00-intro.md).

File: Makefile (hunk 1 above)
Comment (acelaya) [reply Makefile:18]: The PHP version your project requires is defined as any other dependency, in the "require" block of the `composer.json` file. This block is the equivalent to `dependencies` in a package.json file, in the same way `require-dev` is the same as `devDependencies`.

https://github.com/hypothesis/wp-hypothesis/blob/main/composer.json#L11-L13

Unfortunately, it is only used to verify other dependencies you install will work with the PHP version you require, but it does not install the PHP binaries themeselves.

File: README.md (hunk 2 above)
Comment (acelaya) [reply README.md:16]: This typically depends on the system. In Mac most of the people uses brew to install PHP and PHP extensions.

In Linux distros you can use the system package manager, but if you want the latest PHP versions you frequently have to add custom repositories.

In Windows, I have no idea 😅

**Review by robertknight — APPROVED** [review]

**Pull request merged**